### ISSUE TITLE
NullReferenceException when using a generic type parameter in XML documentation

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#324]
Given a generic type parameter, the `UseAliasesInsteadOfConcreteTypeAnalyzer` will throw a `NullReferenceException` on this line:

```csharp
var typeName = symbol.MetadataName;
var namespaceName = symbol.ContainingNamespace.Name;
```

This happens because `ContainingNamespace` is `null`. The exact line of code it occurs at is this (which is part of an XML doc fragment:

```csharp
 <see cref=\"Nullable{T}.Value\"/>
```

Apparently this triggers our inspection as well due to a `TypeArgumentListSyntax` on `{T}`. We should look if it happens for this too:

```csharp
class MyClass<T>
{
}
```

**Comment by Vannevelj** [comment]
This has been rewritten and so far no problem has been found yet. See #348
